pkg/distributor/distributor_test.go

```diff
@@ -3548,17 +3686,23 @@ func prepare(t testing.TB, cfg prepConfig) ([]*Distributor, []mockIngester, []*p
 		},
 		HeartbeatTimeout:     60 * time.Minute,
 		ReplicationFactor:    rf,
-		ZoneAwarenessEnabled: len(cfg.ingesterZones) > 0,
-	}, ingester.IngesterRingKey, ingester.IngesterRingKey, log.NewNopLogger(), nil)
+		ZoneAwarenessEnabled: cfg.totalZones() > 1,
+	}, ingester.IngesterRingKey, ingester.IngesterRingKey, logger, nil)
 	require.NoError(t, err)
 	require.NoError(t, services.StartAndAwaitRunning(context.Background(), ingestersRing))
 
-	test.Poll(t, time.Second, cfg.numIngesters, func() interface{} {
+	test.Poll(t, time.Second, cfg.totalIngesters(), func() interface{} {
 		return ingestersRing.InstancesCount()
 	})
 
 	factory := ring_client.PoolInstFunc(func(inst ring.InstanceDesc) (ring_client.PoolClient, error) {
-		return ingestersByAddr[inst.Addr], nil
+		for i := range ingesters {
+			ing := &ingesters[i] // Take a pointer, so we don't copy the sync.Mutex in the mockIngester
```

Comment:
[non blocking] In a follow up PR we may consider changing `mockIngester` to be a pointer everywhere.

Reply:
I opened https://github.com/grafana/mimir/pull/7223